.ci/Jenkinsfile

```diff
@@ -95,6 +97,72 @@ pipeline {
         }
       }
     }
+    stage('Release') {
+      options { skipDefaultCheckout() }
+      // TODO: for testing purposes
+      //when {
+      //  tag pattern: 'v\\d+\\.\\d+\\.\\d+', comparator: 'REGEXP'
+      //}
+      stages {
+        stage('BuildPublish') {
+          failFast false
+          matrix {
+            agent {label "${PLATFORM}"}
+            options { skipDefaultCheckout() }
+            axes {
+              axis {
+                name 'PLATFORM'
+                values 'arm', 'ubuntu-18 && immutable'
+              }
+            }
+            stages {
+              stage('Dist') {
+                steps {
+                  withGithubNotify(context: "Dist-${PLATFORM}") {
+                    deleteDir()
+                    unstash 'source'
+                    withGoEnv(){
+                      dir("${BASE_DIR}/apm-lambda-extension"){
+                        cmd(label: 'make dist', script: 'make dist')
+                      }
+                    }
+                  }
+                }
+              }
+              stage('Publish') {
+                steps {
+                  withGithubNotify(context: "Publish-${PLATFORM}") {
+                    withAWSEnv(secret: 'secret/observability-team/ci/service-account/apm-aws-lambda', forceInstallation: true, version: '2.4.10') {
+                      dir("${BASE_DIR}/apm-lambda-extension"){
+                        cmd(label: 'make publish-in-all-aws-regions', script: 'make publish-in-all-aws-regions')
+                        stash(includes: "*${PREFIX_ARN_FILE}", name: "arn-${isArm() ? 'arm' : 'amd'}")
+                        stash(allowEmpty: true, name: "arn-${isArm() ? 'arm' : 'amd'}", useDefaultExcludes: false)
+                      }
+                    }
+                  }
+                }
+                post {
+                  always {
+                    archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/apm-lambda-extension/.regions")
+                  }
+                }
+              }
+            }
+          }
+        }
+        stage('Release Notes') {
+          steps {
+            withGhEnv(forceInstallation: true, version: '2.4.0') {
+              dir("${BASE_DIR}/apm-lambda-extension"){
+                unstash "arn-arm"
+                unstash "arn-amd"
```

Comment:
This is the trick to converge the outcome from the publishing stage for the two supported architectures (ARM and AMD64)